jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/CaseServiceImplTest.java

```diff
@@ -2236,4 +2238,415 @@ public void testStartExpressionCaseWithCaseFile() {
+            logger.error("Unexpected error {}", e.getMessage(), e);
+            fail("Unexpected exception " + e.getMessage());
+        } finally {
+            if (caseId != null) {
+                caseService.cancelCase(caseId);
+            }
+        }
+    }
+    
+    @Test
+    public void testUserTaskCaseDataItemWithRestrictions() {
+        Map<String, OrganizationalEntity> roleAssignments = new HashMap<>();
+        roleAssignments.put("owner", new UserImpl("john"));
+        roleAssignments.put("participant", new UserImpl("mary"));
+
+        Map<String, Object> data = new HashMap<>();
+        data.put("contactInfo", "main street 10, NYC");
+        
+        Map<String, List<String>> accessRestrictions = new HashMap<>();
+        accessRestrictions.put("contactInfo", Arrays.asList("owner"));
+        
+        CaseFileInstance caseFile = caseService.newCaseFileInstanceWithRestrictions(deploymentUnit.getIdentifier(), USER_TASK_CASE_P_ID, data, roleAssignments, accessRestrictions);
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), USER_TASK_STAGE_CASE_P_ID, caseFile);
+        assertNotNull(caseId);
+        assertEquals(FIRST_CASE_ID, caseId);
+        try {
+            CaseInstance cInstance = caseService.getCaseInstance(caseId, true, false, false, false);
+            assertNotNull(cInstance);
+            assertEquals(FIRST_CASE_ID, cInstance.getCaseId());
+
+            Map<String, Object> caseData = cInstance.getCaseFile().getData();
+            assertNotNull(caseData);
+            assertEquals(1, caseData.size());            
+            assertEquals("main street 10, NYC", caseData.get("contactInfo"));
+            
+            identityProvider.setName("mary");
+            cInstance = caseService.getCaseInstance(caseId, true, false, false, false);
+            assertNotNull(cInstance);
+            assertEquals(FIRST_CASE_ID, cInstance.getCaseId());
+            // mary should not see any data yet
+            caseData = cInstance.getCaseFile().getData();
+            assertNotNull(caseData);
+            assertEquals(0, caseData.size());
+            
+            identityProvider.setName("john");
+            Collection<CaseStageInstance> activeStages = caseRuntimeDataService.getCaseInstanceStages(caseId, true, new QueryContext());
+            assertNotNull(activeStages);
+            assertEquals(1, activeStages.size());
+
+            caseService.addDataToCaseFile(caseId, "dataComplete", true, "owner", "participant");
+
+            activeStages = caseRuntimeDataService.getCaseInstanceStages(caseId, true, new QueryContext());
+            assertNotNull(activeStages);
+            assertEquals(0, activeStages.size());
+            
+            cInstance = caseService.getCaseInstance(caseId, true, false, false, false);
+            assertNotNull(cInstance);
+            assertEquals(FIRST_CASE_ID, cInstance.getCaseId());
+            // mary should not see any data yet
```

Comment:
Just a question, why is there a comment with Mary when we are logged as John?

Reply:
because I really like Mary :)

Reply:
I mean I understand that as a John you check that he sees all items and then you log in as Mary and there is a comment "mary should not see any data yet" - so she sees only one item, contact info is hidden for her. But why is this comment there when we check if john sees everything - I am clearly missing something :)

Reply:
it was simply copy paste thingy